### PR TITLE
Workshop overhaul (opening sections)

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
## fixes KILTProtocol/ticket#2956

Overhauls the first parts of the workshop section to incorporate changes suggested by the community and a general overhaul and test.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [x] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: docs/develop/03_workshop/03_overview.md [hunk 1]
```diff
@@ -3,21 +3,28 @@ id: overview
...
-But only the <span className="label-role attester">Attester</span> is required to own KILTs since they have to pay for storing the attestation on chain.
-The <span className="label-role verifier">Verifier</span> only needs to query the KILT blockchain to ensure that the attestation is still valid and was not revoked.
-The <span className="label-role claimer">Claimer</span> is not required to query the blockchain, but they might do so to check whether their credential is still valid or if the Attester has revoked it in the meantime.
+-   A <span className="label-role claimer">Claimer</span> is an actor that claims to posses certain credentials, abilities, or other attributes.
+-   An <span className="label-role attester">Attester</span> is an actor that verifies the claims of a <span className="label-role claimer">Claimer</span>.
+-   A <span className="label-role verifier">Verifier</span> is an actor that asks for proof of a claim.
+
+For the workshop you'll play all three roles.
+
+In a real world use case, these actors would be different people and services, which this workshop simulates by using different folders for each service. Each actor typically performs different roles:
```
Comment (Dudleyneedham) [inline docs/develop/03_workshop/03_overview.md]:
```suggestion
In a real world use case, these actors would be different people and services, which this workshop simulates by using different folders for each service.
Each actor typically performs different roles:
```

File: docs/develop/03_workshop/03_overview.md [hunk 2]
```diff
@@ -34,14 +41,26 @@ participant B as KILT Blockchain
...
-Thus, it is important to require this.
-We will [explain CTypes in more detail](./04_attester/03_ctype.md) in a later chapter.
+The <span className="label-role verifier">Verifier</span> requests a presentation from the <span className="label-role claimer">Claimer</span> for a specific required CType. Without a specific CType, the presentation is meaningless.
+
+<!-- TODO: Find out more link… -->
+
 A presentation is derived from a credential and does not need to contain all attributes.
-A <span className="label-role claimer">Claimer</span> could choose to hide their address from their passport if the <span className="label-role verifier">Verifier</span> is only interested in their age.
+
+After the request, the <span className="label-role claimer">Claimer</span> can choose to hide elements of their credentials that aren't relevant to the claim. For example, hide their address from their ID if the <span className="label-role verifier">Verifier</span> is only interested in their age.
```
Comment (Dudleyneedham) [inline docs/develop/03_workshop/03_overview.md]:
```suggestion
After the request, the <span className="label-role claimer">Claimer</span> can choose to hide elements of their credentials that aren't relevant to the claim.
For example, hide their address from their ID if the <span className="label-role verifier">Verifier</span> is only interested in their age.
```

File: docs/develop/03_workshop/03_overview.md [hunk 3]
```diff
@@ -53,3 +72,21 @@ participant B as KILT Blockchain
...
+2. The document is ready, but the Embassy doesn't need all the information in the document. The embassy wants to know if a traveller has sufficient funds, but they don't need to know any transaction details. So, the traveller redacts or hides these details while presenting.
+3. The traveller presents the document to the embassy.
+4. The embassy verified the document's authenticity by comparing its hash value with the one on their internal system or a decentralized ledger.
+5. Since they trust the Attester (in this case, the bank that attested the bank statement), they approved the visa application.
+
+:::tip Summary
+
+As you can see, the Embassy didn't need to trust the Claimer directly in this system. They trusted the Attester, whom they had previously worked with or respected due to their position. And with that trust, they granted the visa.
```
Comment (Dudleyneedham) [inline docs/develop/03_workshop/03_overview.md]:
```suggestion
As you can see, the Embassy didn't need to trust the Claimer directly in this system.
They trusted the Attester, whom they had previously worked with or respected due to their position.
And with that trust, they granted the visa.
```

File: docs/develop/03_workshop/03_overview.md [hunk 4]
```diff
@@ -53,3 +72,21 @@ participant B as KILT Blockchain
...
+
+Even though this process emerged due to the trust in the Attester, the Attester was not involved in the second stage, so they were unaware of it. Privacy was achieved with distributed trust.
```
Comment (Dudleyneedham) [inline docs/develop/03_workshop/03_overview.md]:
```suggestion
Even though this process emerged due to the trust in the Attester, the Attester was not involved in the second stage, so they were unaware of it.
Privacy was achieved with distributed trust.
```

File: docs/develop/03_workshop/03_overview.md (hunk 3 above)
Comment (Dudleyneedham) [reply docs/develop/03_workshop/03_overview.md]: Not sure but would be interesting to note that the attester/issuer doesn't know that the holder/claimer has used the credential in anyway.

**Review by Dudleyneedham — APPROVED** [review]
LGTM. Minor comments but overall looks good.

File: docs/develop/03_workshop/03_overview.md [hunk 5]
```diff
@@ -53,3 +74,23 @@ participant B as KILT Blockchain
...
+2. The document is ready, but the Embassy doesn't need all the information in the document. The embassy wants to know if a traveler has sufficient funds, but they don't need to know any transaction details. The traveler redacts or hides these details while presenting.
+3. The traveler presents the document to the embassy.
+4. The embassy verified the document's authenticity by comparing its hash value with the one on their internal system or a decentralized ledger.
+5. Since they trust the Attester (in this case, the bank that attested the bank statement), they approved the visa application.
+
+:::tip Summary
+
+As you can see, the Embassy didn't need to trust the Claimer directly in this system.
+They trust the Attester, whom they had previously worked with, or respect due to their position.
+And with that trust, they grant the visa with no knowledge of what the Claimer has used the credential for.
```
Comment (ChrisChinchilla) [inline docs/develop/03_workshop/03_overview.md:92]: @Dudleyneedham 

> https://github.com/KILTprotocol/docs/pull/284#discussion_r1384859800

Does this represent what you meant?

Comment (Dudleyneedham) [reply docs/develop/03_workshop/03_overview.md:92]: Yes :)

**Pull request merged**